docs/ansible_turbo_mode.rst

```diff
@@ -0,0 +1,188 @@
+
+- The ``cloud.common`` collection (https://github.com/ansible-collections/cloud.common)
+- `AIOHTTP <https://docs.aiohttp.org/en/stable/>`_ - Asynchronous HTTP Client/Server for asyncio and Python.
+
+
+Installation
+------------
+
+You can install ``cloud.common`` collection using following command::
+
+    # ansible-galaxy collection install cloud.common
+
+You can install ``aiohttp`` using following command::
+
+    # pip install aiohttp
+
+
+Current situation without Ansible Turbo mode
+============================================
+
+The traditional execution flow of an Ansible module includes the following steps:
+
+- Upload of a ZIP archive with the module and its dependencies
+- Execution of the module
+- Ansible collects the results once the script is finished
+
+These steps happen for each task of a playbook, and on every host.
+
+Most of the time, the execution of a module is fast enough for
+the user. However, sometime the module requires significant amount of time,
+just to initialize itself. This is a common situation with the API based modules.
+
+A classic initialization involves the following steps:
+
+- Load a Python library to access the remote resource (via SDK)
+- Open a client
+    - Load a bunch of Python modules.
+    - Request a new TCP connection.
+    - Create a session.
+    - Authenticate the client.
+
+All these steps are time consuming and the same operations will be running again and again.
+
+For instance, here:
+
+- ``import openstack``: takes 0.569s
+- ``client = openstack.connect()``: takes 0.065s
+- ``client.authorize()``: takes 1.360s,
+
+These numbers are from test running against VexxHost public cloud.
+
+In this case, it's a 2s-ish overhead per task. If the playbook
+comes with 10 tasks, the execution time cannot go below 20s.
+
+How Ansible Turbo Module improve the situation
+==============================================
+
+``AnsibleTurboModule`` is actually a class that inherites from
+the standard ``AnsibleModule`` class that your modules probably
+already use.
+The big difference is that when an module starts, it also spawns
+a little Python daemon. If a daemon already exists, it will just
+reuse it.
+All the module logic is run inside this Python daemon. This means:
+
+- Python modules are actually loaded one time
+- Ansible module can reuse an existing authenticated session.
+
+How can I enable ``AnsibleTurboModule``?
+========================================
+
+If you are a collection maintainer and want to enable ``AnsibleTurboModule``, you can
+follow these steps.
+Your module should inherit from ``AnsibleTurboModule``, instead of ``AnsibleModule``.
+
+.. code-block:: python
+
+  from ansible_module.turbo.module import AnsibleTurboModule as AnsibleModule
+
+You can also use the ``functools.lru_cache()`` decorator to ask Python to cache
+the result of an operation, like a network session creation.
+
+Finally, if some of the dependeded libraries are large, it may be nice
```

Comment:
a code block as an example would be nice (even if it is clearly described)

Reply:
OK.